controllers/upgrade_handlers.go

```diff
@@ -18,39 +18,125 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// handleUpgrade orchestrate main upgrade steps
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	// log and update status to in-progress
+	r.Log.Info("Starting handleUpgrade")
+
+	// Pre-pivot steps
+
+	// backup with OADP
+	r.Log.Info("Starting backup")
+	backupRequeue, backupStatus, err := r.BackupRestore.ReconcileBackup(ctx, ibu.Spec.OADPContent)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("error msg from backup: %s, will requeue after: %s", backupStatus.Message, backupRequeue.RequeueAfter))
+		return backupRequeue, err
+	}
+	switch backupStatus.Status {
+	case backuprestore.BackupCompleted:
+		// only proceed when BackupCompleted
+		r.Log.Info(fmt.Sprintf("Backup completed with msg %s", backupStatus.Message))
+	default:
+		// backup in progress
+		utils.SetStatusCondition(&ibu.Status.Conditions,
+			utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade),
+			utils.ConditionReasons.InProgress,
+			metav1.ConditionFalse,
+			backupStatus.Message,
+			ibu.Generation)
+		return backupRequeue, nil
+	}
 
 	utils.ExecuteChrootCmd(utils.Host, "mount /sysroot -o remount,rw")
 	stateRootRepo := fmt.Sprintf("/host/ostree/deploy/rhcos_%s/var", ibu.Spec.SeedImageRef.Version)
 
-	// TODO: Pre-pivot steps
-	//
-	// TODO: Call r.BackupRestore.ExportRestoresToDir to write restore CRs to the new stateroot.
-	//       If there is invalid type of error returned, we should update the ibu status and
-	//       requeue with short interval.
-	//
-	err := r.ExtraManifest.ExportExtraManifestToDir(ctx, ibu.Spec.ExtraManifests, stateRootRepo)
+	r.Log.Info("Writing OadpConfiguratio CRs into new stateroot")
+	err = r.BackupRestore.ExportOadpConfigurationToDir(ctx, stateRootRepo, backuprestore.OadpNs)
 	if err != nil {
-		r.Log.Error(err, "Failed to export extra manifests")
-		return ctrl.Result{}, err
+		msg := "failed to export OadpConfiguratio CRs"
+		r.Log.Error(err, msg)
+		utils.SetStatusCondition(&ibu.Status.Conditions,
```

Comment:
If we want to declare upgrade failure in the conditions, don't return the err back to k8s.
If it's an unexpected/transient error, don't set the conditions and let k8s retry.

Reply:
Also, if we decide we are done with this stage (no more retry or requeue), both conditions need to be set, whether success or failure.

Reply:
We are not requeue-ing on any errors and i brought back the statues that were previously there